src/conversion.rs

```diff
@@ -109,6 +113,12 @@ pub trait IntoPyObject<'py>: Sized {
         let list = crate::types::list::try_new_from_iter(py, &mut iter);
         list.map(Bound::into_any)
     }
+
+    /// The output type of [`owned_sequence_into_pyobject`] and [`borrowed_sequence_into_pyobject`]
+    #[cfg(feature = "experimental-inspect")]
+    #[doc(hidden)]
+    const SEQUENCE_OUTPUT_TYPE: TypeHint =
+        TypeHint::subscript(&PyList::TYPE_HINT, &[Self::OUTPUT_TYPE]);
```

Comment:
Would have been nice to set it to `None` but sadly I have not easily found how to circumvent const expression limits to make it work